### PR TITLE
Disassembled sub_6DE52C. 

### PR DESCRIPTION
**Pull request by aserv**
Added in one new structure relevent to the function, renamed values used by the function to be more descriptive.

### CODE REVIEW

File: src/ride/ride.c [hunk 1]
```diff
@@ -6171,9 +6170,47 @@ void ride_update_max_vehicles(int rideIndex)
...
+	}
+
+	//Same colour per train, different from all existing rides of this type if possible
+	ride->colour_scheme_type = RIDE_COLOUR_SCHEME_ALL_SAME;
+	int remaining = 0xC8;		//Not sure why 200
+	rct_ride* ride_list = RCT2_ADDRESS(RCT2_ADDRESS_RIDE_LIST, rct_ride);
+	uint16 colourIndex;
+	rct_ride* rideTemp;
+
+getRandomColorScheme:
```
Comment (duncanspumpkin) [inline src/ride/ride.c]: I'm not the biggest fan of making this a goto. You could make it a for loop that tries 200 times and if it is succesful returns after setting the colour and if unsuccessful return with colourIndex 0.

File: src/ride/ride.c [hunk 2]
```diff
@@ -6171,9 +6170,47 @@ void ride_update_max_vehicles(int rideIndex)
...
+	if (--remaining != 0) {
+		colourIndex = ((uint16)*colourCount * (uint16)(uint8)scenario_rand()) >> 8;	//High bit of a 8bit x 8bit mul operation
+		rct_vehicle_colour v_colour = colour[colourIndex].vehicle_colour;
+		for (int index = 0; index < 255; index++) {	//Check to see if any rides of this subtype already have this color scheme
+			rideTemp = &ride_list[index];
+			if (rideTemp->type != 255 && rideTemp != ride && ride->subtype == rideTemp->subtype && v_colour.vehicle_colour == rideTemp->vehicle_colours[0].vehicle_colour) {
```
Comment (duncanspumpkin) [inline src/ride/ride.c]: There is far too much in this one line. I'd prefer it if you broke it up first checking to make sure the ride is valid, then checking its not the current ride then...

**Comment by duncanspumpkin** [comment]
I think it might be more useful to keep body and trim as separate addresses in this function. But I'm not that fussed @IntelOrca what do you think?

**Comment by IntelOrca** [comment]
I would probably remove the union and use a function to compare two vehicle colour structs (maybe change to an equals operator overload when we change to C++). But this probably isn't a big deal at the moment.

**Comment by duncanspumpkin** [comment]
Could you rebase this and then it can be merged in.

**Comment by duncanspumpkin** [comment]
Damn something has gone wrong there were no changes merged.

**Comment by janisozaur** [comment]
It looked ok in the git log graph, just that there were some merges. Make sure to retrigger travis if need be.

**Comment by duncanspumpkin** [comment]
I'm not sure what went wrong. Each commit is fine but the merge is empty :(. I've reverted the empty merge but I think all thats done is make the commit history longer.

**Pull request merged**